### DISCUSSION TITLE
Deye Microinverter - Cloud-free

### DISCUSSION THREAD

**Discussion by PV-Joe** [davidrapan/ha-solarman#133] [Show and tell]
**This repository aims to document how to use Deye microinverters entirely local only and cloud free.**

So far, the following devices are known to work like this:

- Deye SUN600G3-EU-230
- Deye SUN-M80G3-EU-Q0
- Deye SUN-M80G4-EU-Q0

**2024-08-20 Update**
Deye has started rolling out firmware updates for the logger that block both ModbusTCP access and prevent the mock cloud from working due to the introduction of TLS. Fortunately, for now it seems that newly bought inverters still come with older logger firmwares that work offline. Also equally fortunately, it seems that downgrading the firmware is possible as well.
However, both of these things may change in the future.

https://github.com/Hypfer/deye-microinverter-cloud-free

**Comment by raspitoaster** [comment]
I use the "dummycloud" to reset the daily production registers - this works very well.
But I still run the old solarman integration in parallel because it features more sensors than the cloud-free MQTT based solution.

**Reply by davidrapan** [reply]
> But I still run the old solarman integration in parallel because it features more sensors than the cloud-free MQTT based solution.

Can you explain little bit more?

**Comment by raspitoaster** [comment]
Of course:

![Sensors - solarman vs dummycloud](https://github.com/user-attachments/assets/7ebac565-aeb0-42f4-a617-fbe547e88807)

**Reply by davidrapan** [reply]
So you are using Stephans integration? I'm not much familiar with that cloud-free plugin so that's why I asked. ;) I wonder what exactly it does to fulfill purpose of that cloud free feature. Maybe it could be done here too? :)

**Reply by raspitoaster** [reply]
Yes, I am still using Stephans integration, but I want to switch to yours because the current one causes issues in my setup.
In parallel, I use the "Deye dummycloud" to set the time&date of the inverter and to reset the counters for the daily production.
Without the "dummycloud", the counters are not reset in the morning if the inverter cannot connect to the Solarman cloud.

**Reply by davidrapan** [reply]
You are already able to set Date & Time using our integration but I'm curious how is done that counter reset but it proly requires some pseudo solarman service which acts as the cloud and thus can't be proly done here.
